app/src/main/java/com/github/bgabriel998/softwaredevproject/map/OSMMap.java

```diff
@@ -189,4 +197,55 @@ private BoundingBox computeArea(ArrayList<POIPoint> points) {
         );
         return new BoundingBox(north.get(), east.get(), south.get(), west.get());
     }
+
+    /**
+     * This method allows to select a point on the map. It allows to pass to the caller
+     * the coordinates of the selected point and it will display a PushPint on the selected
+     * point.
+     *
+     * @param listener      listener for the long press.
+     * @param pointUpdater  used to pass the selected point to the caller.
+     */
+    public void enablePinOnClick(Runnable listener, Consumer<Point> pointUpdater) {
+
+        Overlay touchOverlay = new Overlay(){
+
+            ItemizedIconOverlay<OverlayItem> anotherItemizedIconOverlay = null;
+
+            @Override
+            public boolean onLongPress(final MotionEvent e, final MapView mapView) {
+
+                final Drawable marker = context.getResources().getDrawable(R.drawable.pushpin_marker);
```

Comment:
getDrawable(int) is deprecated and should be replaced by ResourcesCompat.getDrawable().
```suggestion
                final Drawable marker = ResourcesCompat.getDrawable(context.getResources(), 
                        R.drawable.pushpin_marker, null);
```